src/main/java/io/reactivex/internal/operators/parallel/ParallelJoin.java

```diff
@@ -240,8 +238,7 @@ void drainLoop() {
 
                     boolean empty = true;
 
-                    for (int i = 0; i < s.length; i++) {
-                        JoinInnerSubscriber<T> inner = s[i];
+                    for (JoinInnerSubscriber<T> inner : s) {
```

Comment:
This was supposed to be an optimization to avoid reading the array length over and over. The original should have been `for (int i = 0; i < n; i++)`.